src/coreclr/jit/assertionprop.cpp

```diff
@@ -5810,6 +5870,35 @@ ASSERT_VALRET_TP Compiler::optGetVnMappedAssertions(ValueNum vn)
     return BitVecOps::UninitVal();
 }
 
+//------------------------------------------------------------------------
+// optGetEdgeAssertions: Given a block and its predecessor, get the assertions
+//                       the predecessor creates for the block.
+//
+// Arguments:
+//      block     - The block to get the assertions for.
+//      blockPred - The predecessor of the block (creating the assertions).
+//
+// Return Value:
+//      The assertions we have about the value number.
+//
+ASSERT_VALRET_TP Compiler::optGetEdgeAssertions(const BasicBlock* block, const BasicBlock* blockPred) const
+{
+    if (blockPred->KindIs(BBJ_COND) && blockPred->FalseTargetIs(block))
+    {
+        return blockPred->bbAssertionOut;
+    }
+
+    if ((blockPred->KindIs(BBJ_ALWAYS) && blockPred->TargetIs(block)) ||
+        (blockPred->KindIs(BBJ_COND) && blockPred->TrueTargetIs(block)))
+    {
+        if (bbJtrueAssertionOut != nullptr)
+        {
+            return bbJtrueAssertionOut[blockPred->bbNum];
+        }
+    }
+    return BitVecOps::UninitVal();
```

Comment:
Seems like this should return an empty BV on failure, though I also think (once the above is restructured to handle all pred block kinds) you should never get here.

Reply:
returned empty BV